internal/featuregates/featuregates.go

```diff
@@ -19,23 +18,39 @@ var DefaultFeatureGates = map[string]bool{
 	"timeTravel": false,
 }
 
-func IsEnabled(ctx context.Context, client client.Client, feature string) bool {
+type FeatureGateStatus struct {
+	FeatureGates map[string]bool
+}
+
+// This method returns a new FeatureGateStatus object
+// that contains the status of the feature gates
+// defined in the ConfigMap in the namespace
+// If there are any errors in retrieving the ConfigMap, or the feature gates
+// are not defined in the ConfigMap, the default values are used
+func NewFeatureGateStatus(client client.Client) (*FeatureGateStatus, error) {
+	fgStatus := &FeatureGateStatus{
+		FeatureGates: make(map[string]bool),
+	}
+
 	cfgMap := &corev1.ConfigMap{}
-	err := client.Get(ctx,
-		types.NamespacedName{Name: FgConfigMapName, Namespace: OperatorNamespace},
-		cfgMap)
-
-	if err != nil {
-		log.Printf("Error fetching feature gates: %v", err)
-	} else {
-		if value, exists := cfgMap.Data[feature]; exists {
-			return value == "true"
+	err := client.Get(context.TODO(), types.NamespacedName{Name: FgConfigMapName,
+		Namespace: OperatorNamespace}, cfgMap)
+	if err == nil {
+		for feature, value := range cfgMap.Data {
+			fgStatus.FeatureGates[feature] = value == "true"
```

Comment:
Do we want to allow for "True" or "TRUE" as well?  This comparison is case-sensitive so it will only match "true".

Reply:
Currently we are only allowing "true" (case sensitive).

Reply:
Okay, fair enough.  We need to keep in mind documenting it though.